### PR TITLE
[Fix #1705] MissingElse: pass correct loc to add_offense

### LINKED ISSUE #1705: Style/MissingElse crash
I'm using master - `0.29.1 (using Parser 2.2.0.3, running on ruby 2.2.0 x86_64-darwin14)`

The `Style/MissingElse` cop causes the offense report to crash:
```
<file>.rb:9:5: C: if condition requires an else-clause.
undefined method `source_line' for #<Parser::Source::Map::Condition:0x007fde525c2140>
<path>/.rvm/gems/ruby-2.2.0@<gemset>/bundler/gems/rubocop-f52bc7e2ec4b/lib/rubocop/formatter/clang_style_formatter.rb:15:in `block in report_file'
```
The fix seems to be to pass `:expression` as the second argument to `add_offense` (instead of `node.location`). I will submit a PR with the fix soon.

I noticed that when an exception occurs during inspection, it's caught and a message is displayed (`An error occurred while <cop> cop was inspecting <file>.`). Should something similar be done during reporting? Or is there another way we could catch an issue like this ahead of time? Is there a type of spec that tests reporting missing for this cop?

### PR DESCRIPTION
**Pull request by gerry3**

### CODE REVIEW

**Comment by bbatsov** [comment]
It'd be nice to add a regression spec for this.

**Comment by gerry3** [comment]
@bbatsov totally agree. As I said on the issue (#1705):

> I noticed that when an exception occurs during inspection, it's caught and a message is displayed (An error occurred while [cop] cop was inspecting [file].). Should something similar be done during reporting? Or is there another way we could catch an issue like this ahead of time? Is there a type of spec that tests reporting missing for this cop?

Can you provide some direction on an existing method for testing the reporting vs inspection and/or how to make one?

**Comment by rrosenblum** [comment]
This would be my fault. I misunderstood what was supposed to be passed into `add_offense` for the location. I was just diagnosing the same/similar issue while working on a new cop. The weird thing is that on the new cop that I was working on, there was a test failure, but there wasn't a failure in the tests for `MissingElse`. 

It looks like alternatively to passing `:expression` to `add_offense`, `node.location.expression` would also work.

**Comment by gerry3** [comment]
@rrosenblum `add_offense(node, :expression, ...)` is the same as `add_offense(node, node.loc.expression, ...)`, but passing `:expression` appears to be idiomatic

File: spec/rubocop/cop/style/missing_else_spec.rb [hunk 1]
```diff
@@ -256,6 +256,11 @@
           expect(cop.messages)
             .to eq(msg)
         end
+
+        it 'highlights' do
```
Comment (gerry3) [inline spec/rubocop/cop/style/missing_else_spec.rb]: this spec failed before my fix

**Comment by gerry3** [comment]
@bbatsov added a spec that failed for the old code

**Comment by gerry3** [comment]
@bbatsov any thoughts on how my commit could've changed code coverage?!

**Comment by bbatsov** [comment]
> @bbatsov any thoughts on how my commit could've changed code coverage?!

Don't see any noticeable changes. You'll need to rebase this PR, though.

**Comment by gerry3** [comment]
@bbatsov rebased

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**